error.go

```diff
@@ -36,6 +53,86 @@ func (e internalError) InternalError() error {
 	return e.error
 }
 
+// Code set the error's translation code.
+func (e Error) Code(code string) Error {
+	e.code = code
+
+	return e
+}
+
+// GetCode get the error's translation code.
+func (e Error) GetCode() string {
+	return e.code
+}
+
+// Params set the error's params.
+func (e Error) Params(params map[string]interface{}) Error {
+	e.params = params
+	return e
+}
+
+// GetParams returns the error's params.
+func (e Error) GetParams() map[string]interface{} {
+	return e.params
+}
+
+// Message set the error's message.
+func (e Error) Message(message string) Error {
+	e.message = message
+	return e
+}
+
+// GetMessage return the error's message.
+func (e Error) GetMessage() string {
+	return e.message
+}
+
+// Error returns the error message.
+func (e Error) Error() string {
+	if len(e.params) == 0 {
+		return e.message
+	}
+
+	res := bytes.Buffer{}
+	_ = template.Must(template.New("err").Parse(e.message)).Execute(&res, e.params)
+
+	return res.String()
+}
+
+// Translate get a translator that must implemented Translator and
+// return translated errors.
+func (e Error) Translate(t Translator) (string, error) {
```

Comment:
I'm thinking to drop the translate methods completely from this library. Since the new `Error` struct and the built-in rules carry/expose the error code information, the translation feature can be done separately.

Reply:
Right, so I remove it.